src/simulator/deck.py

```diff
@@ -0,0 +1,84 @@
+import numpy as np
+
+# Values for blackjack cards from Ace through King for each suit
+CARD_VALUES = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 10, 10] * 4
+CARDS_IN_DECK = 52
+
+
+class EndOfChuteException(Exception):
+    """
+    Signals the Chute has ended and we need to reshuffle the decks
+    """
+
+
+class Deck:
+    """
+    Simulates a dealer's deck(s)
+    """
+    def __init__(self, num_decks=6, reshuffle_perc=0.8, random_seed=None):
+        """
+        :param num_decks: Number of decks the dealer uses to shuffle
+        :param reshuffle_perc: The proportion through all cards where the
+                               decks are re-shuffled
+        :param random_seed: A random seed to initialize the random number
+                            generator, so we can repeat experiments
+        """
+        self.num_decks = num_decks
+        self.reshuffle_at = int(reshuffle_perc * num_decks * CARDS_IN_DECK)
+
+        if random_seed is not None:
+            np.random.seed(random_seed)
+
+        # Put these here so PyCharm would stop yelling at me
+        self.cards = None
+        self.card_index = None
```

Comment:
When we init you could probably just set the index to 0. Might make more sense than None